stream_alert/rule_processor/handler.py

```diff
@@ -49,24 +57,38 @@ def __init__(self, context, enable_alert_processor=True):
         self.sinker = StreamSink(self.env)
 
         # Instantiate a classifier that is used for this run
-        self.classifier = StreamClassifier(config=config)
+        self.classifier = StreamClassifier(config=self.config)
 
         self.enable_alert_processor = enable_alert_processor
         self._failed_record_count = 0
         self._processed_size = 0
         self._alerts = []
 
+        # Create a dictionary to hold parsed payloads by log type.
+        # Firehose needs this information to send to its corresponding
+        # delivery stream.
+        self.categorized_payloads = defaultdict(list)
+
+        # Firehose initialization
+        firehose_config = self.config['global'].get(
```

Comment:
Can you just create the `self.firehose_client` object on the `init` and then have a method to handle instantiation during the `run` (instead of putting the if/else block in the init)